spinalcordtoolbox/utils/shell.py

```diff
@@ -98,6 +99,38 @@ def display_viewer_syntax(files, colormaps=[], minmax=[], opacities=[], mode='',
         printv(cmd + '\n', verbose=1, type='info')
 
 
+class SCTArgumentParser(argparse.ArgumentParser):
```

Comment:
Today in stupid OOP tricks for 100 points: You might consider splitting this into parts:

```
class AutoHelpParser(argparse.ArgumentParser):
   def error(): ...
...

class SCTArgumentParser(AutoHelpParser, argparse.ArgumentParser):
   def __init__(...):
       self.add_argument_group("OPTIONAL ARGUMENTS") ......
...
```

That would make using it for #2676 clearer because you could add:


```
class SCTArgumentParser(VerbosityParser, AutoHelpParser, argparse.ArgumentParser):
   def __init__(...):
       self.add_argument_group("OPTIONAL ARGUMENTS") ......
...
```


But you could also split it when you get to #2676. Or just not? It's easy to go overboard with OOP.

Reply:
Interesting! Admittedly, I haven't explored multiple inheritance very deeply, so I like that you brought it up. Gets me to think about new things. :D

My only reservation is that it makes things a little more complicated to understand. And, that added complexity might not be worth it, because `AutoHelpParser` and `VerbosityParser` would only be used within `SCTArgumentParser` (since we only need one argparser across all of our scripts.).

Reply:
Totally, always glad to share roadshow programming tricks.